docs/integration/driver/display/st7789.rst

```diff
@@ -0,0 +1,248 @@
+a command or pixel data to the controller via SPI or parallel bus. Typically these are implemented by calling the appropriate SDK library functions on the given platform.
+
+Prerequisites
+-------------
+
+There are no prerequisites.
+
+Configuring the driver
+----------------------
+
+Enable the ST7789 driver support in lv_conf.h, by cmake compiler define or by KConfig
+
+.. code:: c
+
+	#define LV_USE_ST7789  1
+
+Usage
+-----
+
+You need to implement two platform-dependent functions:
+
+.. code:: c
+
+	/* Send short command to the LCD. This function shall wait until the transaction finishes. */
+	int32_t my_lcd_send_cmd(lv_display_t *disp, uint8_t *cmd, size_t cmd_size, uint8_t *param, size_t param_size)
+	{
+		...
+	}
+
+	/* Send large array of pixel data to the LCD. If necessary, this function has to do the byte-swapping. This function can do the transfer in the background. */
+	int32_t my_lcd_send_color(lv_display_t *disp, uint8_t *cmd, size_t cmd_size, uint8_t *param, size_t param_size)
+	{
+		...
+	}
+
+The only difference between the ``my_lcd_send_cmd()`` and ``my_lcd_send_color()`` functions is that ``my_lcd_send_cmd()`` is used to send short commands and it is expected
+complete the transaction when it returns (in other words, it should be blocking), while ``my_lcd_send_color()`` is only used to send pixel data, and it is recommended to use
+DMA to transmit data in the background. More sophisticated methods can be also implemented, like queing transfers and scheduling them in the background.
+
+Please note that while display flushing is handled by the driver, it is the user's responsibility to call ``lv_display_flush_ready()``
+when the color transfer completes. In case of a DMA transfer this is usually done in a transfer ready callback.
+
+.. note::
+	While it is acceptable to use a blocking implementation for the pixel transfer as well, performance will suffer.
+
+.. tip::
+	Care must be taken to avoid sending a command while there is an active transfer going on in the background. It is the user's responsibility to implement this either
+	by polling the hardware, polling a global variable (which is reset at the end of the transfer), or by using a semaphore or other locking mechanism.
+
+Please also note that the driver does not handle the draw buffer allocation, because this may be platform-dependent, too. Thus you need to allocate the buffers and assign them
+to the display object as usual by calling ``lv_display_set_draw_buffers()``.
+
+The driver can be used to create multiple displays. In such a configuration the callbacks must be able to distinguish between the displays. Usually one would
+implement a separate set of callbacks for each display. Also note that the user must take care of arbitrating the bus when multiple devices are connected to it.
+
+Example
+-------
+
+This is an example implementation on an STM32F746 using STM32CubeIDE and the ST HAL libraries.
+
+.. code:: c
+
```

Comment:
Let's create a docs page for the generic mipi driver as well and move the example there too.